Flakey unittests for client

**Describe the bug**
GHA job test-client-python is flakey.
### Error
```
>           raise MaxRetryError(_pool, url, error or ResponseError(cause))
E           urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8080): Max retries exceeded with url: /v1/graphql (Caused by ProtocolError('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')))
```
**To Reproduce**
open a PR and wait for *test-client-python* to run.

**Screenshots**
![Screenshot 2024-07-25 at 12 07 53 PM](https://github.com/user-attachments/assets/13bbe62c-8dde-4931-b180-29fdd58ef7f4)

--- a/client/python/cryoet_data_portal/Makefile
+++ b/client/python/cryoet_data_portal/Makefile
@@ -1,6 +1,6 @@
 .PHONY: test-infra
 test-infra:
-	cd tests && docker compose up -d
+	cd tests && docker compose up -d --wait --wait-timeout 300
 	cd ./tests/test_infra/; ./seed_moto.sh
 
 .PHONY: coverage

--- a/client/python/cryoet_data_portal/tests/docker-compose.yml
+++ b/client/python/cryoet_data_portal/tests/docker-compose.yml
@@ -49,6 +49,8 @@ services:
       HASURA_GRAPHQL_ENABLE_CONSOLE: "true"
       HASURA_GRAPHQL_DEV_MODE: "true"
       HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
+    healthcheck:
+      test: timeout 1s bash -c ':> /dev/tcp/127.0.0.1/8080' || exit 1
     depends_on:
       db:
         condition: service_healthy

